packages/next/server/utils.ts

```diff
@@ -15,53 +16,19 @@ export function cleanAmpPath(pathname: string): string {
   return pathname
 }
 
-export type Disposable = () => void
-export type Observer<T> = {
-  next(chunk: T): void
-  error(error: Error): void
-  complete(): void
-}
-export type RenderResult = (observer: Observer<string>) => Disposable
-
-export function resultFromChunks(chunks: string[]): RenderResult {
-  return ({ next, complete, error }) => {
-    let canceled = false
-    process.nextTick(() => {
-      try {
-        for (const chunk of chunks) {
-          if (canceled) {
-            return
-          }
-          next(chunk)
-        }
-      } catch (err) {
-        if (!canceled) {
-          canceled = true
-          error(err)
-        }
-      }
+export type RenderResult = Observable<string>
 
-      if (!canceled) {
-        canceled = true
-        complete()
-      }
-    })
-
-    return () => {
-      canceled = true
-    }
-  }
+export function mergeResults(results: Array<RenderResult>): RenderResult {
+  // @ts-ignore
+  return Observable.prototype.concat.call(...results)
 }
 
-export function resultToChunks(result: RenderResult): Promise<string[]> {
-  return new Promise((resolve, reject) => {
-    const chunks: string[] = []
-    result({
-      next: (chunk) => {
-        chunks.push(chunk)
-      },
-      error: (error) => reject(error),
-      complete: () => resolve(chunks),
-    })
+export async function resultsToString(
+  results: Array<RenderResult>
+): Promise<string> {
+  const chunks: string[] = []
+  await mergeResults(results).forEach((chunk) => {
```

Comment:
Why `await` here?

Reply:
`Observable<T>.forEach` returns a promise that indicates when the observable is complete

Reply:
`Observable.forEach` returns a promise that resolves when the observable is finished